src/web/status.rs

```diff
@@ -0,0 +1,151 @@
+use super::cache::CachePolicy;
+use crate::{
+    db::Pool,
+    utils::spawn_blocking,
+    web::{error::AxumResult, match_version_axum},
+};
+use axum::{
+    extract::{Extension, Path},
+    http::header::ACCESS_CONTROL_ALLOW_ORIGIN,
+    response::IntoResponse,
+    Json,
+};
+
+pub(crate) async fn status_handler(
+    path: Path<(String, String)>,
+    pool: Extension<Pool>,
+) -> impl IntoResponse {
+    async fn inner(
+        Path((name, req_version)): Path<(String, String)>,
+        Extension(pool): Extension<Pool>,
+    ) -> AxumResult<impl IntoResponse> {
+        let (version, id) = match_version_axum(&pool, &name, Some(&req_version))
+            .await?
+            .exact_name_only()?
+            .into_parts();
+
+        let rustdoc_status: bool = spawn_blocking({
+            move || {
+                Ok(pool
+                    .get()?
+                    .query_one(
+                        "SELECT releases.rustdoc_status
+                         FROM releases
+                         WHERE releases.id = $1
+                        ",
+                        &[&id],
+                    )?
+                    .get("rustdoc_status"))
+            }
+        })
+        .await?;
+
+        Ok((
+            Extension(CachePolicy::NoStoreMustRevalidate),
+            [(ACCESS_CONTROL_ALLOW_ORIGIN, "*")],
+            Json(serde_json::json!({
+                "version": version,
+                "doc_status": rustdoc_status,
+            })),
+        ))
+    }
+    (
+        Extension(CachePolicy::NoStoreMustRevalidate),
```

Comment:
Why do we have to duplicate the caching & header definitions?

Reply:
Mistake, they should only have been applied outside.